ds3-sdk/src/main/java/com/spectralogic/ds3client/metadata/MetadataReceivedListenerImpl.java

```diff
@@ -0,0 +1,254 @@
+            aclEntries.add(aclEntry);
+        } catch (final Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    //set the last modified time saved on server
+    private void setModifiedTime(final String filePath, final Long modifiedTime) {
+        try {
+            final Path file = Paths.get(filePath);
+            final BasicFileAttributes attr =
+                    Files.readAttributes(file, BasicFileAttributes.class);
+            final FileTime ft = FileTime.fromMillis(modifiedTime);
+            Files.setLastModifiedTime(file, ft);
+        } catch (final Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    //set the permissions saved on server
+    private void setPermissions(final String filePath, final String permissions) {
+        try {
+            final Path file = Paths.get(filePath);
+            final Set<PosixFilePermission> perms =
+                    PosixFilePermissions.fromString(permissions);
+/*            FileAttribute<Set<PosixFilePermission>> attr =
+                    PosixFilePermissions.asFileAttribute(perms);*/
+            Files.setPosixFilePermissions(file, perms);
+        } catch (final Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    //set owner and group name on local from the black perl server
+    private void setOwnerNGroup(final String filePath, final String ownerName, final String groupName) {
+        try {
+            final Path file = Paths.get(filePath);
+            final UserPrincipal owner = file.getFileSystem().getUserPrincipalLookupService()
+                    .lookupPrincipalByName(ownerName);
+            Files.setOwner(file, owner);
+            final GroupPrincipal group =
+                    file.getFileSystem().getUserPrincipalLookupService()
+                            .lookupPrincipalByGroupName(groupName);
+            Files.getFileAttributeView(file, PosixFileAttributeView.class)
+                    .setGroup(group);
+        } catch (final Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void setFileTimes(final String filePath, final String creationTime, final String lastModifiedTime, final String lastAccessedTime) {
+        try {
+            final BasicFileAttributeView attributes = Files.getFileAttributeView(Paths.get(filePath), BasicFileAttributeView.class);
+            final FileTime timeCreation = FileTime.fromMillis(Long.parseLong(creationTime));
+            final FileTime timeModified = FileTime.fromMillis(Long.parseLong(lastModifiedTime) + 60000);
+            final FileTime timeAccessed = FileTime.fromMillis(Long.parseLong(lastAccessedTime));
+            attributes.setTimes(timeModified, timeAccessed, timeCreation);
+            final File f = new File(filePath);
+            f.setLastModified(Long.parseLong(lastModifiedTime) + 60000);
+
+        } catch (final Exception e) {
+            e.printStackTrace();
```

Comment:
Exceptions should not be handled in this way. At a minimum it should be logged with a slf4j logger, or wrapped in a runtime exception and re-thrown.